charms/precise/clearwater-bono/lib/config_script

```diff
@@ -15,6 +15,8 @@ enum_server=
 
 # Apply new configuration
 home_domain=$(config-get zone)
+[ "$relation-name" != "cscf" ] || [ "$(relation-get public-address)" = "" ] || sprout_hostname=$(relation-get public-address)
```

Comment:
It is the right behavior, because public-address is set to the sprout cluster name.  (Relations are between services, not between units - did I get that terminology right?)

See https://github.com/Metaswitch/clearwater-juju/blob/dnsaas/charms/precise/clearwater-sprout/hooks/pcscf-relation-joined#L4:

    relation-set public-address=sprout.$(config-get zone)